### ISSUE TITLE
Github private repository

### ISSUE THREAD

**Issue by ritmas** [ansible/ansible-lint#4060]
### Discussed in https://github.com/ansible/ansible-lint/discussions/4053

<div type='discussions-op-text'>

<sup>Originally posted by **ritmas** February 29, 2024</sup>
How should I run `ansible-lint` Github action on repository which `requirements.yml` file contains private Github repository next to public ones?

Now such attempt fails due to lack of permissions:
```bash
| ERROR    [WARNING]: - {PRIVATE-REPO} was NOT installed successfully: - command
| /usr/bin/git clone ssh://git@github.com/{PRIVATE-REPO}
| {PRIVATE-REPO} failed in directory /root/.ansible/tmp/ansible-
| local-163xakt__cf/tmpvpbdv44y (rc=128) - Cloning into '{PRIVATE-REPO}'...
| git@github.com: Permission denied (publickey).  fatal: Could not read from
| remote repository.  Please make sure you have the correct access rights and the
| repository exists.
| ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.

```</div>

**Comment by ritmas** [comment]
Adding extra step to setup Git config for accessing private repositories solved the issue:
```yaml
jobs:
  build:
    name: Ansible Lint # Naming the build is important to use it as a status check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      
      - name: Prepare Git for Github
        shell: bash
        run: |
          git config --global url."https://${{ secrets.GITHUB_TOKEN }}@github.com".insteadOf "https://github.com"

      - name: Run ansible-lint
        uses: ansible/ansible-lint@main # or version tag instead of 'main'
```
